presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -401,6 +405,27 @@ private void processNewSources()
                 }
             }
 
+            for (int index = operators.size() - 1; index >= 0; index--) {
+                if (operators.get(index).isFinished()) {
+                    // close and remove all operators up to and including this operator
+                    List<Operator> finishedOperators = this.operators.subList(0, index + 1);
+                    Throwable throwable = closeAndDestroyOperators(finishedOperators);
+                    finishedOperators.clear();
+                    if (throwable != null) {
+                        throwIfUnchecked(throwable);
+                        throw new RuntimeException(throwable);
+                    }
+                    // ensure new root operator is finishing
```

Comment:
`// Finish the next operator, which is now the first operator.`

I feel root operator is an ambiguous term here.